python/lsst/ip/isr/crosstalk.py

```diff
@@ -23,17 +23,441 @@
 Apply intra-detector crosstalk corrections
 """
 import numpy as np
+from astropy.table import Table
 
 import lsst.afw.math
-import lsst.afw.table
 import lsst.afw.detection
 from lsst.pex.config import Config, Field, ChoiceField, ListField
 from lsst.pipe.base import Task
 
-__all__ = ["CrosstalkConfig", "CrosstalkTask", "subtractCrosstalk", "writeCrosstalkCoeffs",
+from lsst.ip.isr import IsrCalib
+
+
+__all__ = ["CrosstalkCalib", "CrosstalkConfig", "CrosstalkTask",
            "NullCrosstalkTask"]
 
 
+class CrosstalkCalib(IsrCalib):
+    """Calibration of amp-to-amp crosstalk coefficients.
+
+    Parameters
+    ----------
+    detector : `lsst.afw.cameraGeom.Detector`, optional
+        Detector to use to pull coefficients from.
+    log : `lsst.log.Log`, optional
+        Log to write messages to.
+    **kwargs :
+        Parameters to pass to parent constructor.
+    """
+    _OBSTYPE = 'CROSSTALK'
+    _SCHEMA = 'Gen3 Crosstalk'
+    _VERSION = 1.0
+
+    def __init__(self, detector=None, **kwargs):
+        self.hasCrosstalk = False
+        self.nAmp = 0
+        self.crosstalkShape = (self.nAmp, self.nAmp)
+
+        self.coeffs = None
+        self.coeffErr = None
+        self.coeffNum = None
+        self.interChip = None
+
+        super().__init__(**kwargs)
+        self.requiredAttributes.update(['hasCrosstalk', 'nAmp', 'coeffs',
+                                        'coeffErr', 'coeffNum', 'interChip'])
+        if detector:
+            self.fromDetector(detector)
+
+    def updateMetadata(self, setDate=False, **kwargs):
+        """Update calibration metadata.
+
+        This calls the base class's method after ensuring the required
+        calibration keywords will be saved.
+
+        Parameters
+        ----------
+        setDate : `bool`, optional
+            Update the CALIBDATE fields in the metadata to the current
+            time. Defaults to False.
+        kwargs :
+            Other keyword parameters to set in the metadata.
+        """
+        kwargs['DETECTOR'] = self._detectorName
+        kwargs['DETECTOR_SERIAL'] = self._detectorSerial
+        kwargs['HAS_CROSSTALK'] = self.hasCrosstalk
+        kwargs['NAMP'] = self.nAmp
+        self.crosstalkShape = (self.nAmp, self.nAmp)
+        kwargs['CROSSTALK_SHAPE'] = self.crosstalkShape
+
+        super().updateMetadata(setDate=setDate, **kwargs)
+
+    def fromDetector(self, detector, coeffVector=None):
+        """Set calibration parameters from the detector.
+
+        Parameters
+        ----------
+        detector : `lsst.afw.cameraGeom.Detector`
+            Detector to use to set parameters from.
+        coeffVector : `numpy.array`, optional
+            Use the geometry in the detector, but supply the
+            coefficients from elsewhere.
```

Comment:
You mean if this is not provided, it uses detector geometry, but if it is, it uses the coefficients in `coeffVector`? Not clear to me as written.